main/recording-history.js

```diff
@@ -0,0 +1,304 @@
+  const recordings = [newRecording, ...recordingHistory.get('recordings', [])];
+  const validRecordings = recordings.filter(({filePath}) => fs.existsSync(filePath));
+  recordingHistory.set('recordings', validRecordings);
+  return validRecordings;
+};
+
+const cleanPastRecordings = () => {
+  const recordings = getPastRecordings();
+  for (const recording of recordings) {
+    fs.unlinkSync(recording.filePath);
+  }
+
+  recordingHistory.set('recordings', []);
+};
+
+const cleanUpRecordingPlugins = usedPlugins => {
+  const recordingPlugins = plugins.getRecordingPlugins();
+
+  for (const pluginName of Object.keys(usedPlugins)) {
+    const plugin = recordingPlugins.find(p => p.name === pluginName);
+    for (const [serviceTitle, persistedState] of Object.entries(usedPlugins[pluginName])) {
+      const service = plugin && plugin.recordServices.find(s => s.title === serviceTitle);
+
+      if (service && service.cleanUp) {
+        service.cleanUp(persistedState);
+      }
+    }
+  }
+};
+
+const handleIncompleteRecording = async recording => {
+  cleanUpRecordingPlugins(recording.plugins);
+  try {
+    await execa(ffmpegPath, [
+      '-i',
+      recording.filePath,
+      '-v',
+      'error',
+      '-f',
+      'null',
+      '-'
+    ]);
+
+    return handleRecording(recording);
+  } catch (error) {
+    return handleCorruptRecording(recording, error.stderr);
+  }
+};
+
+const handleRecording = async recording => {
+  addRecording({
+    filePath: recording.filePath,
+    name: recording.name,
+    date: recording.date
+  });
+
+  return showDialog({
+    title: 'Kap didn\'t shut down correctly',
+    detail: 'Looks like Kap crashed during a recording. We were able to locate the file and it appears to be playable.',
+    buttons: [
+      'Close',
+      {
+        label: 'Open in Finder',
+        action: () => shell.showItemInFolder(recording.filePath)
+      },
+      {
+        label: 'Open in Editor',
+        action: () => openEditorWindow(recording.filePath, {recordingName: recording.name})
+      }
+    ]
+  });
+};
+
+const knownErrors = [{
+  test: error => error.includes('moov atom not found'),
+  fix: async filePath => {
+    try {
+      const outputPath = tempy.file({extension: 'mp4'});
+
+      await execa(ffmpegPath, [
+        '-i',
+        filePath,
+        '-vcodec',
+        'copy',
+        '-acodec',
+        'copy',
+        '-movflags',
+        'faststart',
+        outputPath
+      ]);
+
+      return outputPath;
+    } catch {
+
+    }
+  }
+}];
+
+const handleCorruptRecording = async (recording, error) => {
+  const options = {
+    title: 'Kap didn\'t shut down correctly',
+    detail: `Looks like Kap crashed during a recording. We were able to locate the file. Unfortunately it appears to be corrupt.\n\n${error}`,
+    cancelId: 0,
+    defaultId: 2,
+    buttons: [
+      'Close',
+      {
+        label: 'Copy Error',
+        action: () => clipboard.writeText(error)
```

Comment:
It's IMHO more readable to not use inline arrow functions when the function body doesn't return anything.

```suggestion
        action: () => {
        	clipboard.writeText(error);
        }
```

Reply:
I assume `error` is the error message, not the error object, so I would name it clearer: `errorMessage`.